lib/internal/Magento/Framework/Backup/BackupInterface.php

```diff
@@ -68,4 +68,19 @@ public function getType();
      * @return $this
      */
     public function setBackupsDir($backupsDir);
+
+    /**
+     * Check if keep files of backup
+     *
+     * @return boolean
+     */
+    public function keepSourceFile();
```

Comment:
unfortunately you cannot add new classes to interfaces (http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/index.html#introduction-of-a-method-to-a-class-or-interface) in this case I would recommend adding a new interface specific for the source file flags.